xonsh/ptk_shell/shell.py

```diff
@@ -76,6 +78,25 @@ def tokenize_ansi(tokens):
     return ansi_tokens
 
 
+def remove_ansi_osc(prompt):
+    """Removes the ANSI OSC escape codes - ``prompt_toolkit`` does not support them.
+    Some terminal emulators - like iTerm2 - uses them for various things.
+
+    See: https://www.iterm2.com/documentation-escape-codes.html
+    """
+
+    osc_tokens = ANSI_OSC_PATTERN.findall(prompt)
+    prompt = re.sub(ANSI_OSC_PATTERN, "", prompt)
```

Comment:
This should be `ANSI_OSC_PATTERN.sub("", prompt)`

Reply:
Fixed.